src/ryu.jl

```diff
@@ -6,8 +6,11 @@ function plain_precision_heuristic(xs::AbstractArray{<:AbstractFloat})
     ys = filter(isfinite, xs)
     precision = 0
     for y in ys
-        b, e10 = Ryu.reduce_shortest(convert(Float32, y))
-        precision = max(precision, -e10)
+        if y ≈ 0.0 # 0 digits for numbers close to 0
```

Comment:
The problem oftentimes arises due 0.0 present as usually `0.0 = 1e-55` (some weird power for floats). Then this would mean that if 0.0 is present, precision would go to 55. It is better to set that precision needed to display 0 should always be 0

Reply:
```julia
julia> 1e-200 ≈ 0.0
false
```
I think you'll be better off keeping a min/max precision and capping the ratio between them.